ee/clickhouse/queries/session_recordings/clickhouse_session_recording_list.py

```diff
@@ -1,119 +1,234 @@
+            timestamp_clause += "\nAND timestamp >= %(event_start_time)s"
+            timestamp_params["event_start_time"] = self._filter.date_from - timedelta(hours=12)
+        if self._filter.date_to:
+            timestamp_clause += "\nAND timestamp <= %(event_end_time)s"
+            timestamp_params["event_end_time"] = self._filter.date_to + timedelta(hours=12)
+        return timestamp_clause, timestamp_params
+
+    def _get_recording_start_time_clause(self) -> Tuple[str, Dict[str, Any]]:
+        start_time_clause = ""
+        start_time_params = {}
+        if self._filter.date_from:
+            start_time_clause += "\nAND start_time >= %(start_time)s"
+            start_time_params["start_time"] = self._filter.date_from
+        if self._filter.date_to:
+            start_time_clause += "\nAND start_time <= %(end_time)s"
+            start_time_params["end_time"] = self._filter.date_to
+        return start_time_clause, start_time_params
+
+    def _get_duration_clause(self) -> Tuple[str, Dict[str, Any]]:
+        duration_clause = ""
+        duration_params = {}
+        if self._filter.recording_duration_filter:
+            if self._filter.recording_duration_filter.operator == "gt":
+                operator = ">"
+            else:
+                operator = "<"
+            duration_clause = "\nAND duration {operator} %(recording_duration)s".format(operator=operator)
+            duration_params = {
+                "recording_duration": self._filter.recording_duration_filter.value,
+            }
+        return duration_clause, duration_params
+
+    def format_event_filter(self, entity: Entity, prepend: str) -> Tuple[str, Dict[str, Any]]:
+        filter_sql, params = format_entity_filter(entity, prepend=prepend, filter_by_team=False)
+        if entity.properties:
+            filters, filter_params = parse_prop_clauses(
+                entity.properties,
+                prepend=prepend,
+                team_id=self._team.pk,
+                allow_denormalized_props=False,
+                has_person_id_joined=True,
+                table_name="filtered_events",
+                person_properties_mode=PersonPropertiesMode.USING_PERSON_PROPERTIES_COLUMN,
+            )
+            filter_sql += f" {filters}"
+            params = {**params, **filter_params}
+
+        return filter_sql, params
+
+    def format_event_filters(self) -> EventFiltersSQL:
+
+        aggregate_select_clause = ""
+        aggregate_having_clause = ""
+        where_conditions = "AND event IN %(event_names)s"
+        # Always include $pageview events so the start_url and end_url can be extracted
+        event_names_to_filter: Set[Union[int, str]] = set(["$pageview"])
+
+        params: Dict = {}
+
+        for index, entity in enumerate(self._filter.entities):
+            if entity.type == TREND_FILTER_TYPE_ACTIONS:
+                action = entity.get_action()
+                for action_step in action.steps.all():
```

Comment:
Does this work given that action steps can have their own property filters as well? Seems like this would discard all of them right?

Reply:
So this just filtering which events get considered in the filters, so it shouldn't filter out any of the properties themselves.